sdks/python/apache_beam/dataframe/partitionings_test.py

```diff
@@ -36,8 +36,8 @@ class PartitioningsTest(unittest.TestCase):
   }).set_index(['shape', 'color', 'size'])
 
   def test_index_is_subpartition(self):
-    ordered_list = [Nothing(), Index([3]), Index([1, 3]), Index(), Singleton()]
-    for loose, strict in zip(ordered_list[:1], ordered_list[1:]):
+    ordered_list = [Singleton(), Index([3]), Index([1, 3]), Index(), Nothing()]
+    for loose, strict in zip(ordered_list[:-1], ordered_list[1:]):
```

Comment:
Note the change to make this `-1`. This test was only verifying `Singleton()` vs `Index([3])` then exiting the loop. The checks between different `Index()` partitionings would have failed, but they weren't being executed.

Reply:
Ack.